caas/kubernetes/provider/bootstrap.go

```diff
@@ -622,130 +512,182 @@ func (c controllerStack) buildStorageSpecForController(statefulset *apps.Statefu
+				TimeoutSeconds:      1,
 			},
-		},
-	})
-
-	// add container API server.
-	containerSpec = append(containerSpec, core.Container{
-		Name: "api-server",
-		// ImagePullPolicy: core.PullIfNotPresent,
-		ImagePullPolicy: core.PullAlways, // TODO(bootstrap): for debug
-		Image:           c.pcfg.GetControllerImagePath(),
-		VolumeMounts: []core.VolumeMount{
-			{
-				Name:      c.pvcNameControllerPodStorage,
-				MountPath: c.pcfg.DataDir,
+			LivenessProbe: &core.Probe{
+				Handler: core.Handler{
+					Exec: probCmds,
+				},
+				FailureThreshold:    3,
+				InitialDelaySeconds: 30,
+				PeriodSeconds:       10,
+				SuccessThreshold:    1,
+				TimeoutSeconds:      5,
 			},
-			{
-				Name:      c.resourceNameVolAgentConf,
-				MountPath: filepath.Join(c.pcfg.DataDir, "agents", ("machine-" + c.pcfg.MachineId), c.fileNameAgentConfMount),
-				SubPath:   c.fileNameAgentConfMount,
+			VolumeMounts: []core.VolumeMount{
+				{
+					Name:      c.pvcNameControllerPodStorage,
+					MountPath: c.pcfg.DataDir,
+				},
+				{
+					Name:      c.pvcNameControllerPodStorage,
+					MountPath: filepath.Join(c.pcfg.DataDir, "db"),
+					SubPath:   "db",
+				},
+				{
+					Name:      c.resourceNameVolSSLKey,
+					MountPath: filepath.Join(c.pcfg.DataDir, c.fileNameSSLKeyMount),
+					SubPath:   c.fileNameSSLKeyMount,
+					ReadOnly:  true,
+				},
+				{
+					Name:      c.resourceNameVolSharedSecret,
+					MountPath: filepath.Join(c.pcfg.DataDir, c.fileNameSharedSecret),
+					SubPath:   c.fileNameSharedSecret,
+					ReadOnly:  true,
+				},
 			},
-			{
-				Name:      c.resourceNameVolSSLKey,
-				MountPath: filepath.Join(c.pcfg.DataDir, c.fileNameSSLKey),
-				SubPath:   c.fileNameSSLKey,
-				ReadOnly:  true,
+		})
+
+		// add container API server.
+		containerSpec = append(containerSpec, core.Container{
+			Name:            "api-server",
+			ImagePullPolicy: core.PullIfNotPresent,
+			// ImagePullPolicy: core.PullAlways, // TODO(bootstrap): for debug
+			Image: c.pcfg.GetControllerImagePath(),
+			Command: []string{
+				"/bin/sh",
 			},
-			{
-				Name:      c.resourceNameVolSharedSecret,
-				MountPath: filepath.Join(c.pcfg.DataDir, c.fileNameSharedSecret),
-				SubPath:   c.fileNameSharedSecret,
-				ReadOnly:  true,
+			Args: []string{
+				"-c",
+				fmt.Sprintf(jujudStartUpSh, jujudCmd),
 			},
-			{
-				Name:      c.resourceNameVolBootstrapParams,
-				MountPath: filepath.Join(c.pcfg.DataDir, c.fileNameBootstrapParams),
-				SubPath:   c.fileNameBootstrapParams,
-				ReadOnly:  true,
+			WorkingDir: jujudToolDir,
+			VolumeMounts: []core.VolumeMount{
+				{
+					Name:      c.pvcNameControllerPodStorage,
+					MountPath: c.pcfg.DataDir,
+				},
+				{
+					Name: c.resourceNameVolAgentConf,
+					MountPath: filepath.Join(
+						c.pcfg.DataDir,
+						"agents",
+						"machine-"+c.pcfg.MachineId,
+						c.fileNameAgentConfMount,
+					),
+					SubPath: c.fileNameAgentConfMount,
+				},
+				{
+					Name:      c.resourceNameVolSSLKey,
```

Comment:
I don't quite understand why we need all these individual mounts.
Why can't we just mount a single config map at DataDir, and then all the files needed are available directly from the config map keys